### PR TITLE
fix(crux-ui): json editor and mapper

### PR DESCRIPTION
**Pull request by peterambrus3**

### CODE REVIEW

**Review by polaroi8d — APPROVED** [review]
Looks okay for me.

File: web/crux-ui/src/components/products/versions/images/edit-image-json.tsx [hunk 1]
```diff
@@ -34,8 +34,12 @@ const EditImageJson = (props: EditImageJsonProps) => {
 
   const throttle = useThrottling(IMAGE_WS_REQUEST_DELAY)
 
+  const [jsonError, setJsonError] = useState<boolean>(false)
+
   const onMergeValues = (_: JsonConfig, local: JsonConfig): JsonConfig => {
-    onPatch(mergeJsonConfigToContainerConfig(config, local))
+    if (!jsonError) {
+      onPatch(mergeJsonConfigToContainerConfig(config, local))
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/images/edit-image-json.tsx:41]: why not merge with the remote? (atm named '_')

File: web/crux-ui/src/models/container.ts [hunk 2]
```diff
@@ -565,50 +565,59 @@ export const mergeJsonConfigToContainerConfig = (
...
-        return {
-          ...prev,
-          args: mergeKeysWithJson(prev.args, it.args),
-          command: mergeKeysWithJson(prev.command, it.command),
-          environment: mergeKeyValuesWithJson(prev.environment, it.environment),
-          volumes: it.volumes?.map(vit => {
-            const volumeIndex = prev.volumes?.map(pv => pv.name).indexOf(vit.name)
-            const id = volumeIndex !== -1 ? prev.volumes[volumeIndex].id : uuid()
+    initContainers:
+      json.initContainers == null
```
Comment (m8vago) [inline web/crux-ui/src/models/container.ts:569]: use `!json.initContainers`

File: web/crux-ui/src/models/container.ts [hunk 3]
```diff
@@ -565,50 +565,59 @@ export const mergeJsonConfigToContainerConfig = (
...
+              const prev = serialized.initContainers[index]
+
+              return {
+                ...prev,
+                args: mergeKeysWithJson(prev.args, it.args),
+                command: mergeKeysWithJson(prev.command, it.command),
+                environment: mergeKeyValuesWithJson(prev.environment, it.environment),
+                volumes: it.volumes?.map(vit => {
+                  const volumeIndex = prev.volumes?.map(pv => pv.name).indexOf(vit.name)
+                  const id = volumeIndex !== -1 ? prev.volumes[volumeIndex].id : uuid()
```
Comment (m8vago) [inline web/crux-ui/src/models/container.ts:584]: `volumeIndex > 0`

File: web/crux-ui/src/models/container.ts (hunk 2 above)
Comment (peterambrus3) [reply web/crux-ui/src/models/container.ts:569]: done, thx

File: web/crux-ui/src/models/container.ts (hunk 3 above)
Comment (peterambrus3) [reply web/crux-ui/src/models/container.ts:584]: done, thx

File: web/crux-ui/src/components/products/versions/images/edit-image-json.tsx (hunk 1 above)
Comment (peterambrus3) [reply web/crux-ui/src/components/products/versions/images/edit-image-json.tsx:41]: we talked about this in person, the scope is bigger

**Review by polaroi8d — APPROVED** [review]

**Review by m8vago — APPROVED** [review]

**Pull request merged**